hack/boilerplate/boilerplate.py

```diff
@@ -73,14 +94,18 @@ def file_passes(filename, refs, regexs):
     f.close()
 
     basename = os.path.basename(filename)
-    extension = file_extension(filename)
+    if generated:
```

Comment:
we have the file read into data here already. Shouldn't we do the twoliner of line 79+80 here?

Reply:
At beginning of func is_generated_file, there is also a judgement to the script that contain "DO NOT EDIT".
I think all the judgement of is_generated_file in one func is more clear.

Reply:
how about pass the data to is_generated_file, avoid read it twice？

Reply:
@sttts Have done, PTAL